internal/checkers/negative_postive.go

```diff
@@ -0,0 +1,200 @@
+		}
+
+	case "Greater":
+		if len(call.Args) < 2 {
+			return nil
+		}
+		a, b := call.Args[0], call.Args[1]
+
+		if isZero(a) && isNotZero(b) {
+			return newUseNegativeDiagnostic(a.Pos(), b.End(), b)
+		}
+
+	case "True":
+		if len(call.Args) < 1 {
+			return nil
+		}
+		expr := call.Args[0]
+
+		a, _, ok1 := isStrictComparisonWith(pass, expr, p(isNotZero), token.LSS, p(isZero)) // a < 0
+		_, b, ok2 := isStrictComparisonWith(pass, expr, p(isZero), token.GTR, p(isNotZero)) // 0 > a
+
+		survivingArg, ok := anyVal([]bool{ok1, ok2}, a, b)
+		if ok {
+			return newUseNegativeDiagnostic(expr.Pos(), expr.End(), survivingArg)
+		}
+
+	case "False":
+		if len(call.Args) < 1 {
+			return nil
+		}
+		expr := call.Args[0]
+
+		a, _, ok1 := isStrictComparisonWith(pass, expr, p(isNotZero), token.GEQ, p(isZero)) // a >= 0
+		_, b, ok2 := isStrictComparisonWith(pass, expr, p(isZero), token.LEQ, p(isNotZero)) // 0 <= a
+
+		survivingArg, ok := anyVal([]bool{ok1, ok2}, a, b)
+		if ok {
+			return newUseNegativeDiagnostic(expr.Pos(), expr.End(), survivingArg)
+		}
+	}
+	return nil
+}
+
+func (checker NegativePositive) checkPositive(pass *analysis.Pass, call *CallMeta) *analysis.Diagnostic {
+	newUsePositiveDiagnostic := func(replaceStart, replaceEnd token.Pos, replaceWith ast.Expr) *analysis.Diagnostic {
+		const proposed = "Positive"
+		return newUseFunctionDiagnostic(checker.Name(), call, proposed,
+			newSuggestedFuncReplacement(call, proposed, analysis.TextEdit{
+				Pos:     replaceStart,
+				End:     replaceEnd,
+				NewText: analysisutil.NodeBytes(pass.Fset, replaceWith),
+			}),
+		)
+	}
+
+	switch call.Fn.NameFTrimmed {
+	case "Greater":
+		if len(call.Args) < 2 {
+			return nil
+		}
+		a, b := call.Args[0], call.Args[1]
+
+		if isNotZero(a) && isZero(b) {
+			return newUsePositiveDiagnostic(a.Pos(), b.End(), a)
+		}
+
+	case "Less":
+		if len(call.Args) < 2 {
+			return nil
+		}
+		a, b := call.Args[0], call.Args[1]
+
+		if isZero(a) && isNotZero(b) {
+			return newUsePositiveDiagnostic(a.Pos(), b.End(), b)
+		}
+
+	case "True":
+		if len(call.Args) < 1 {
+			return nil
+		}
+		expr := call.Args[0]
+
+		a, _, ok1 := isStrictComparisonWith(pass, expr, p(isNotZero), token.GTR, p(isZero)) // a > 0
+		_, b, ok2 := isStrictComparisonWith(pass, expr, p(isZero), token.LSS, p(isNotZero)) // 0 < a
+
+		survivingArg, ok := anyVal([]bool{ok1, ok2}, a, b)
+		if ok {
+			return newUsePositiveDiagnostic(expr.Pos(), expr.End(), survivingArg)
+		}
+
+	case "False":
+		if len(call.Args) < 1 {
+			return nil
+		}
+		expr := call.Args[0]
+
+		a, _, ok1 := isStrictComparisonWith(pass, expr, p(isNotZero), token.LEQ, p(isZero)) // a <= 0
+		_, b, ok2 := isStrictComparisonWith(pass, expr, p(isZero), token.GEQ, p(isNotZero)) // 0 >= a
+
+		survivingArg, ok := anyVal([]bool{ok1, ok2}, a, b)
+		if ok {
+			return newUsePositiveDiagnostic(expr.Pos(), expr.End(), survivingArg)
+		}
+	}
+	return nil
+}
+
+func isStrictComparisonWith(
```

Comment:
analogue of neighbour `isComparisonWith`

Reply:
I like be the functions like this, like the isNotZero and co.

The code is now starting to use function from other checkers, it's OK. But, I would like to suggest an improvement.

What do you think about them moving out of each checker into a "lib" in the same folder ?

So they could be more visible, and would help people to contribute. *Also they could be tested*.

I mean yesterday, I started working on something for checking require.TestingT, the I found later an existing function doing it.

TL;D; I'm OK with current PR, I'm talking about moving then in a new issue/PR

Reply:
@Antonboom I will resolve this one, but I would like to get your point of view about the idea of moving methods like this into a common file, so xor, isZero, isNotZero and co and are not spread among all files and added where they are needed the first time.

I'm asking because I wrote a few code before noticing they were already available in another file, with a different name than the one I chose

Reply:
> not spread among all files and added where they are needed the first time.

yep, I followed this approach while code base was not so significant 

1. I did not want to move helpers in separate package, because the most of them requires `Pass` and could not be tested separately from analyzer. 
2. Also I am afraid of "Helpers Anti-Pattern" 

> of moving methods like this into a common file

looks like compromise, something like
- `checkers/helpers_len.go`
- `checkers/helpers_bool.go`
- `checkers/helpers_error.go`
?

I can’t figure out what’s better and that’s why I left everything as it is.

Therefore, I’m open to any proposals, just take into account the points above 👍

Reply:
Yes, I was talking about having them in the same package than now.

I like your idea of spitting them by context, so contributor won't be lost in a huge "helpers.go"

Reply:
I opened https://github.com/Antonboom/testifylint/issues/84 to continue the discussion there.